docs/dask.rst

```diff
@@ -53,3 +62,132 @@ a progress bar:
+functions to all the spectra or all the spatial images in a cube:
+:meth:`~spectral_cube.DaskSpectralCube.apply_function_parallel_spectral` and
+:meth:`~spectral_cube.DaskSpectralCube.apply_function_parallel_spatial`. By default, these methods
+take functions that apply to individual spectra or images, but this can be quite slow for large
+spectral cubes. If possible, you should consider supplying a function that can accept 3-d cubes
+and operate on all spectra or image slices in a vectorized way.
+
+To demonstrate this, we will read in a large CASA dataset with 623 channels and 768x768 pixels in
+the image plane::
+
+    >>> large = SpectralCube.read('large_spectral_cube.image', format='casa_image', use_dask=True)  # doctest: +SKIP
+    >>> large  # doctest: +SKIP
+    DaskVaryingResolutionSpectralCube with shape=(623, 768, 768) and unit=Jy / beam:
+    n_x:    768  type_x: RA---SIN  unit_x: deg    range:   290.899389 deg:  290.932404 deg
+    n_y:    768  type_y: DEC--SIN  unit_y: deg    range:    14.501466 deg:   14.533425 deg
+    n_s:    623  type_s: FREQ      unit_s: Hz     range: 216201517973.483 Hz:216277445708.200 Hz
+
+As an example, we will apply sigma clipping to all spectra in the cube. Note that there is a method
+to do this (:meth:`~spectral_cube.DaskSpectralCube.sigma_clip_spectrally`) but for the purposes of
+demonstration, we will set up the function ourselves and apply it with
+:meth:`~spectral_cube.DaskSpectralCube.apply_function_parallel_spectral`. We will use the
+:func:`~astropy.stats.sigma_clip` function from astropy::
+
+    >>> from astropy.stats import sigma_clip
+
+By default, this function returns masked arrays, but to apply this to our
+spectral cube, we need it to return a plain Numpy array with NaNs for the masked
+values. In addition, the original function tends to return warnings which we want to
+silence, so we can do this here too::
+
+    >>> import warnings
+    >>> import numpy as np
+    >>> def sigma_clip_with_nan(*args, **kwargs):
+    ...     with warnings.catch_warnings():
+    ...         warnings.simplefilter('ignore')
+    ...         return sigma_clip(*args, axis=0, **kwargs).filled(np.nan)
+
+The ``axis=0`` is so that if the function is passed a cube, it will still work properly.
+
+Let's now call :meth:`~spectral_cube.DaskSpectralCube.apply_function_parallel_spectral`, including the
+``save_to_tmp_dir`` option mentioned previously to force the calculation and the storage of the result
+to disk::
+
+    >>> clipped_cube = large.apply_function_parallel_spectral(sigma_clip_with_nan, sigma=3,
+    ...                                                       save_to_tmp_dir=True)  # doctest: +SKIP
+    [########################################] | 100% Completed |  1min 42.3s
+
+The ``sigma`` argument is passed to the ``sigma_clip_with_nan`` function. We now call this
+again but specifying that the ``sigma_clip_with_nan`` function can also take cubes, using
+the ``accepts_chunks=True`` option::
```

Comment:
is this true?  I thought `sigma_clip` would change behavior if not specifically run on a spectrum?  Have you verified the results are identical?

Reply:
The ``axis=0`` in the wrapped function is what makes it work ok

Reply:
I've tried to clarify this